site/docs/reference/Connectors/capture-connectors/sqlserver.md

```diff
@@ -0,0 +1,277 @@
+
+## Supported versions and platforms
+
+This connector supports SQL Server 2017 and later on major cloud providers,
+as well as self-hosted instances.
+Setup instructions are provided for the following platforms:
+
+* [Self-hosted SQL Server](#setup-self-hosted-sql-server)
+* [Azure SQL Database](#setup-azure-sql-database)
+* [Amazon RDS for SQL Server](#setup-amazon-rds-for-sql-server)
+* [Google Cloud SQL for SQL Server](#setup-google-cloud-sql-for-sql-server)
+
+## Prerequisites
+
+To capture change events from SQL Server tables using this connector, you need:
+
+* For each table to be captured, a primary key should be specified in the database.
+If a table doesn't have a primary key, you must manually specify a key in the associated Flow collection definition while creating the capture.
+[See detailed steps](#specifying-flow-collection-keys).
+
+* [CDC enabled](https://learn.microsoft.com/en-us/sql/relational-databases/track-changes/enable-and-disable-change-data-capture-sql-server?view=sql-server-ver16)
+on the database and the individual tables to be captured.
+(This creates *change tables* in the database, from which the connector reads.)
+
+* A **watermarks table**. The watermarks table is a small “scratch space” to which the connector occasionally writes a small amount of data to ensure accuracy when backfilling preexisting table contents.
+
+* A user role with:
+  * `SELECT` permissions on the CDC schema and the schemas that contain tables to be captured.
+  * `SELECT` and `UPDATE` permissions on the watermarks table
+
+To meet these requirements, follow the steps for the your hosting type.
+
+* [Self-hosted SQL Server](#setup-self-hosted-sql-server)
+* [Azure SQL Database](#setup-azure-sql-database)
+* [Amazon RDS for SQL Server](#setup-amazon-rds-for-sql-server)
+* [Google Cloud SQL for SQL Server](#setup-google-cloud-sql-for-sql-server)
+
+### Setup: Self-hosted SQL Server
+
+1. Connect to the server and issue the following commands:
+
+```sql
+USE <database>;
+--Enable CDC for the database.
+EXEC sys.sp_cdc_enable_db;
+--Create user and password for use with the connector.
+CREATE LOGIN flow_capture WITH PASSWORD = 'secret';
+CREATE USER flow_capture FOR LOGIN flow_capture;
+/*Grant the user permissions on the CDC schema and schemas with data.
+This assumes all tables to be captured are in the default schema, `dbo`.
+Add similar queries for any other schemas that contain tables you want to capture.*/
+GRANT SELECT ON SCHEMA :: dbo TO flow_capture;
+GRANT SELECT ON SCHEMA :: cdc TO flow_capture;
+--Create the watermarks table, grant permissions, and enable CDC.
+CREATE TABLE dbo.flow_watermarks(slot INTEGER PRIMARY KEY, watermark TEXT);
+GRANT SELECT, UPDATE ON dbo.flow_watermarks TO flow_capture;
+EXEC sys.sp_cdc_enable_table @source_schema = 'dbo', @source_name = 'flow_watermarks', @role_name = 'flow_capture', @capture_instance = 'dbo_flow_watermarks';
```

Comment:
Yesterday I just copy-pasted you the setup sequence instructions we're currently using our test database initialization. Now that I'm reading through this in the context of user-facing instructions, I think it would be clearer if we changed the last few lines to something like:

```
-- Add some data to the watermarks table to ensure proper connector initiation.
INSERT INTO dbo.flow_watermarks VALUES (0, 'dummy-value');
-- Enable CDC on the watermarks table. You should run this query for all other
-- tables you intend to capture as well.
EXEC sys.sp_cdc_enable_table @source_schema = 'dbo', @source_name = 'flow_watermarks', @role_name = 'flow_capture';
```

(The most important bit is mentioning that they'll need to run `sys.sp_cdc_enable_table` on all the other tables they intend to capture. I figure that moving that command to the end makes that fact a bit more noticeable. Removing the capture instance name is a no-op here since the default is `<schema>_<table>` but it seems marginally easier for the user to generalize the query to other tables if they only have to change the source table name in one spot.)